modules/apps/asset/asset-info-impl/src/main/java/com/liferay/asset/info/internal/item/provider/AssetEntryInfoItemFieldSetProviderImpl.java

```diff
@@ -115,41 +194,63 @@ public InfoFieldSet getInfoFieldSet(String itemClassName) {
 		}
 	}
 
-	private String _getCategoryNames(AssetEntry assetEntry) {
-		Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
+	private Set<AssetVocabulary> _getAssetVocabularies(AssetEntry assetEntry) {
+		List<AssetCategory> assetCategories = assetEntry.getCategories();
+		Set<AssetVocabulary> assetVocabularies = new HashSet<>();
 
-		return ListUtil.toString(
-			assetEntry.getCategories(),
-			new Accessor<AssetCategory, String>() {
+		for (AssetCategory assetCategory : assetCategories) {
+			AssetVocabulary assetVocabulary =
+				_assetVocabularyLocalService.fetchAssetVocabulary(
+					assetCategory.getVocabularyId());
 
-				@Override
-				public String get(AssetCategory assetCategory) {
-					String title = assetCategory.getTitle(locale);
+			assetVocabularies.add(assetVocabulary);
+		}
 
-					if (Validator.isNull(title)) {
-						return assetCategory.getName();
-					}
+		return assetVocabularies;
+	}
 
-					return title;
-				}
+	private String _getCategoryNames(AssetEntry assetEntry, long vocabularyId) {
+		Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
```

Comment:
We are still getting the locale this way, which we can't assume will always be present (for example in tests it's often not).
The only solution I can think for this would be to go ahead and do the conversion from a list of categories to a list of categoryNames closer to the frontend and not within these providers. We can do it after this pull, but we should not forget about it.